File: examples-sdk/java/010-basic-websocket-chat/Main.java
// 010-basic-websocket-chat — Java SDK equivalent
// Compile: vil compile --from java --input 010-basic-websocket-chat/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("websocket-chat", 8080);
        ServiceProcess chat = new ServiceProcess("chat");
        chat.endpoint("GET", "/", "index");
        chat.endpoint("GET", "/ws", "ws_handler");
        chat.endpoint("GET", "/stats", "stats");
        server.service(chat);
        server.compile();
    }
}

File: examples-sdk/csharp/010-basic-websocket-chat/Main.cs
// 010-basic-websocket-chat — C# SDK equivalent
// Compile: vil compile --from csharp --input 010-basic-websocket-chat/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("websocket-chat", 8080);
var chat = new ServiceProcess("chat");
chat.Endpoint("GET", "/", "index");
chat.Endpoint("GET", "/ws", "ws_handler");
chat.Endpoint("GET", "/stats", "stats");
server.Service(chat);
server.Compile();
